aten/src/ATen/TensorImpl.cpp

```diff
@@ -89,6 +91,34 @@ IntList TensorImpl::strides() const {
   return strides_;
 }
 
+int64_t TensorImpl::numel() const {
+#ifdef DEBUG
+  int64_t n = 1;
+  for (auto s : sizes()) {
+    n *= s;
+  }
+  AT_ASSERT(n == numel_);
+#endif
+  return numel_;
+}
+
+void TensorImpl::refresh_contiguous() {
+  is_contiguous_ = true;
```

Comment:
I think it's better to just write once (i.e. store the computed value as a second variable and set it once at the end); implementing it this way makes it more complicated to put watches or even to printf debug.